lldp/rx.c

```diff
@@ -406,21 +406,24 @@ void rxProcessFrame(struct port *port, struct lldp_agent *agent)
 			}
 		}
 
-		/* rx per lldp module */
-		LIST_FOREACH(np, &lldp_mod_head, lldp) {
-			if (!np->ops || !np->ops->lldp_mod_rchange)
-				continue;
-
-			err = np->ops->lldp_mod_rchange(port, agent, tlv);
-
-			if (!err)
-				tlv_stored = true;
-			else if (err == TLV_ERR) {
-				frame_error++;
-				free_unpkd_tlv(tlv);
-				goto out;
-			}
-		}
+                if (!tlv_stored) {
```

Comment:
I disagree to this - we should reference count the TLV data, I think.  What if multiple modules reference the same TLV.  For one thing, I'm not confident that there aren't current cases in the codebase where we need this (for example, I don't generally do much testing with DCBX).
Is there a case where we know this currently is happening?  Or is it just a change "to be safe"?

Reply:
If there is a situation currently where multiple modules reference the same TLV then I think it would crash because there is no reference counting so they would both try to free the TLV. Therefore I was updating the code to simply avoid that ever being a possibility.

It's a "to be safe" change that I saw when trying to debug the other memory issues. However, as it is more controversial and I don't have much time or the various setups to investigate it further, I will drop it for now.